jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -197,242 +189,92 @@
         false, // 0x7f DEL is illegal
     };
 
+    private static final boolean[] ENCODE_PATH_NEEDS_ENCODING;
+
     private URIUtil()
     {
     }
 
+    static
+    {
+        ENCODE_PATH_NEEDS_ENCODING = new boolean[128];
+        // Special characters
+        for (char c: "%?;#\"'<> [\\]^`{|}".toCharArray())
+            ENCODE_PATH_NEEDS_ENCODING[c] = true;
+        // control characters
+        ENCODE_PATH_NEEDS_ENCODING[0x7f] = true;
+        for (int i = 0; i < 0x20; i++)
+            ENCODE_PATH_NEEDS_ENCODING[i] = true;
```

Comment:
I believe SPACE 0x20 should be encoded too.

Reply:
Already covered in the special chars above